org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/desktop/TabbedDesktop.java

```diff
@@ -299,13 +299,13 @@ else if(Window.REPLACE.equals(window.getAttribute(Window.INSERT_POSITION_KEY)))
 			}
 
 			if (refTab == null)
-				windowContainer.addWindow(tabPanel, title, true, null);
+				windowContainer.addWindow(tabPanel, title, true, new DecorateInfo("Window"));
 			else
 				windowContainer.replace(refTab, window, title);
 
 		}
 		else {
-	    	windowContainer.addWindow(tabPanel, title, true, null);
+	    	windowContainer.addWindow(tabPanel, title, true, new DecorateInfo("Window"));
```

Comment:
hmm .. can we be sure that this "Window" and not "Report", "Process", "Info" or "Form" ?

Reply:
Yeah, I'm not sure how to check what type of window here. Need deeper testing. Just set default window icon to make sure the tab has icon.

Reply:
that could be misleading to always use the "Window" icon.

Reply:
Hi @hengsin . i Tried to trace back method showEmbedded and i think it's impossible to detect which is the Window child class extends Window class. If we want to set icon properly, we need to pass window attribute for the icon to set correctly. lets take WAcctViewer that trigger this method. when showWindow is called, we cannot information this is custom form. on DPViews, the icon for Account Info is hardcoded on the class. I just take Window as default icon to make sure the tab has icon.

Reply:
Hi @uthadehikaru - as you said "we need to pass window attribute" - I tested the following patch to open the "Workflow Activities" form from the "Activities" dashboard with success:
[pr2568PR0.patch.txt](https://github.com/user-attachments/files/17992158/pr2568PR0.patch.txt)

I think the same can be done for the other types.

NOTE aside: It would be good to define constants somewhere for the "Window", "Report", "Process", "Info" and "Form" - and use the constants throughout the system instead of those hardcoded values

Reply:
Hi @CarlosRuiz-globalqss , that is what i mean. we need to send attribute like patch you sent. but it's not answer @hengsin concern, how can be sure what type of the window if no attribute defined. can we use window icon or something else? or just leave it empty?

Reply:
Regardless, we should integrate the patch from Carlos above. For existing case where we can't determine the type of the window, we can use a default icon but that shouldn't be the "Window" icon since that would be misleading. My preference would be to just leave it empty.

Reply:
Can we define default icon for window that opened from this method? I tried to traceback and my conclusion is window that opened here is mostly from custom form. we can set DecorateInfo on All custom form of idempiere core class but custom form from plugin will be empty. I suggest this icon as default icon
https://static.vecteezy.com/system/resources/previews/000/575/451/original/windows-tab-icon-vector.jpg
WDYT

Reply:
Yes, I'm ok with that.